src/dropdown/dropdown.js

```diff
@@ -264,9 +247,57 @@ angular.module('ui.bootstrap.dropdown', ['ui.bootstrap.position'])
       if (!dropdownCtrl.dropdownMenu) {
         dropdownCtrl.dropdownMenu = element;
       }
+
+      var dropdown, appendToBody,
+          dropdownMenu = element;
+
+      scope.$watch(dropdownCtrl.isOpen, function( isOpen ) {
```

Comment:
I think this $watch needs to be in the controller - it should be noted that `dropdownCtrl.isOpen` will evaluate immediately when the linking function is run, so this watch is basically
```js
scope.$watch(false, function(isOpen) {
...
});
```
which is not what we want.